code/framework/extension/src/main/java/io/cattle/platform/extension/impl/ExtensionManagerImpl.java

```diff
@@ -134,12 +139,24 @@ public synchronized void start() {
                 ExtensionList<?> extensionList = getExtensionListInternal(key);
                 extensionList.inner.clear();
                 extensionList.inner.addAll(getList(key));
+                ExtensionMap<String, Object> extensionMap = getExtensionMapInternal(key);
+                extensionMap.inner.clear();
+                extensionMap.inner.putAll(getMap(key));
             }
 
             started = true;
         }
     }
 
+    protected synchronized Map<String, Object> getMap(String key) {
```

Comment:
Hm. TBH, I cant see why getList is synchronized either. @sonchang do you see a reason why getList() would be synchronized and/or why this method would need to be?